src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -226,6 +255,14 @@ public List<TResult> ToList()
 
             public int GetCount(bool onlyIfCheap)
             {
+                if (!onlyIfCheap)
```

Comment:
The reason for this is obscure and would likely benefit from being commented on. Without context this looks like pointless busy work that should be deleted to improve efficiency.